test/files/neg/t5887.scala

```diff
@@ -3,9 +3,28 @@ trait TheOldCollegeTry {
 
   // was: value isDefinedAt is not a member of Int
   // now: required: PartialFunction[Throwable,?]
-  //def f = try ??? catch 22
+  def f = try ??? catch 22
 
   def g = try 42
 
   def h = List("x") map (s => try { case _ => 7 })
+
+  def j = try ??? catch (_ => 42)
```

Comment:
This one caught my eye.  Using `case` we get a warning:

```scala
scala> try ??? catch { case _ => 42 }
                            ^
       warning: This catches all Throwables. If this is really intended, use `case _ : Throwable` to clear this warning.
val res2: Int = 42
```

Could get the same warning for your example?  Not a merge blocker IMO.

Reply:
"It's just a function."

How about:
```
warning: This is a function. If this is really intended, it should be enabled by importing language.functionalProgramming
```

Reply:
Catching OutOfMemoryExceptions (and then allocating) is lint worthy, IMO...

Reply:
Apparently my previous joke is not a joke. I think you'd have to lint the feature. But if a catch function is just a dangerous niche feature, then it seems to me that it doesn't pull its weight. Maybe it should lint `try`.
```
Did you mean `Try`?
```